scripts/main.js

```diff
@@ -78,462 +78,459 @@ let keysObj2 = {
+      keysContainer2.style.justifyContent = "center";
+      keysContainer2.style.alignItems = "center";
+      keyboardLayout.appendChild(keysContainer2);
+    }
+
+    // SPECIAL BUTTONS INJECTOR
+    let btnInjector = (array) => {
+
+      // button objects with special functions
+      const capsLock = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-arrow-alt-up",
+        },
+        operator: () => {
+          let btn = document.querySelectorAll(".keysContainer1 button");
+          let indicatorBtn = document.querySelector(".fa-arrow-alt-up div");
+
+          for (let i = 0; i < btn.length; i++) {
+
+            if (/[a-zA-Z]/.test(btn[i].innerText) && btn[i].innerText != switcher2) {
+              let isLowerCase = true;
+
+              if (/[a-z]/.test(btn[i].innerHTML)) {
+                btn[i].innerText = btn[i].innerText.toUpperCase();
+                indicatorBtn.style.background = " rgb(16, 230, 16)";
+                isLowerCase = false;
+              }
+
+              if (isLowerCase) {
+                indicatorBtn.style.background = "rgba(218, 218, 218, 0.6)";
+                btn[i].innerText = btn[i].innerText.toLowerCase();
+              }
+            }
+          }
+        }
+      }
+
+      const backSpace = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-backspace"
+        },
+        operator: () => {
+          console.clear();
+
+          // if caret already at 0 index, do NOTHING!
+          if (input.selectionStart === 0) return;
+
+          // get access to the input value
+          let txt = input.value;
+
+          // set start index conditionally:
+          //  one position behind the caret if no selection is made, or
+          //  no modification, if a selection range is made
+          let startIdx =
+            input.selectionStart === input.selectionEnd
+              ? input.selectionStart - 1
+              : input.selectionStart;
+
+          let endIdx = input.selectionEnd;
+
+          // declare a "pre-selection" slice of text, a "middle" which will be the deleted portion, and a "post-selection" slice of text
+          let pre = txt.slice(0, startIdx);
+          let mid = txt.slice(startIdx, endIdx);
+          let post = txt.slice(endIdx);
+
+          // concat pre and post, leaving out the selected middle
+          let newTxt = pre.concat(post);
+
+          // update the input with the new chars
+          input.value = newTxt;
+
+          // set caret with offset to account for deleted characters
+          input.setSelectionRange(startIdx, startIdx);
+
+          // not sure what this does
+          // prevent the caret to jump if it was positioned at the beginning of the text input
+          // if (caret === 0) {
+          //     input.setSelectionRange(0, 0);
+          // }
+        },
+      };
```

Comment:
Your code threw a syntax error. I think this curly bracket might be the reason.

Reply:
Removed the hanging bracket (not sure why that didn't break things while I was working.)  Please confirm but this should be working OK now.